### PR TITLE
feat(feg_relay): inject sgw C teid from Feg Relay

### LINKED ISSUE #9625: [Feg relay] inject TEID to inbound roaming requests
Currently for inbound roaming, MME is the originator of TEIDs. MME is the one in charge to generate teids and keep them unique within AGW instance.

Since we are moving to a centralized TEID model, MME will not do this task anymore. Instead the Create Session Request and Create Bearer Request will not be sending the TEID value (neither for control or data plane). Instead, FEG relay will inject that value (coming from orc8r) during the request.

For that reason we need to modify current FEG relay to inject control and user plane TEIDs for inbound roaming request on s8_proxy request

Some requirements are:
- Unique teids will have to be extracted from indexers (probably directory_record index)
- We need to do this just for s8_proxy, but we need to create a generic code in case we need to add similar feature in the future for other request

### PR DESCRIPTION
**Pull request by uri200**
Signed-off-by: Oriol Batalla <obatalla@fb.com>

## Summary

This PR adds Feg Relay teid injection capabilities 

The teid will be generated at directoryd. Directoryd will generate a random number and make sure that number is not in the indexer. 

Feg Relay will write that teid in CreateBearerRequest

This feature only targets Control Plane. User Plane will be modified in future request

## Test Plan

```
./build -g
./build -c
```

Tested locally
```
# build orc8r and configure one network
cd /Users/obatalla/magma/lte/gateway/python/integ_tests/federated_tests
fab build_orc8r
# fab clear_orc8r
fab configure_orc8r


# bootstrap gateway
${MAGMA_ROOT}/orc8r/tools/scripts/bootstrap.bash 5f8c12e7-14d0-4a67-b35f-a432ccccfc4d

cd /tmp/magma_protos/ && \
protoc \
    -I . \
    -I orc8r/protos/prometheus/ \
    --descriptor_set_out out.protoset \
    --include_imports \
    orc8r/cloud/go/services/state/protos/indexer.proto orc8r/protos/directoryd.proto

# request teid
grpcurl \
    -insecure \
    -key /tmp/magma_protos/gateway.key \
    -cert /tmp/magma_protos/gateway.crt \
    -authority directoryd-controller.magma.test \
    -protoset /tmp/magma_protos/out.protoset \
    -d '{"networkID": "feg_lte_test"}' \
    localhost:7443 \
    magma.orc8r.DirectoryLookup/GetNewSgwCTeid

# result of the request
{
  "teid": "3478320140"
}

```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/s8_relay.go [hunk 1]
```diff
@@ -98,3 +111,24 @@ func (s S8RelayRouter) getS8Client(c context.Context, imsi string) (protos.S8Pro
...
+	if err != nil {
+		return 0, fmt.Errorf("Couldnt retrieve GatewayIdentity: %s", err)
+	}
+	teid, err := directoryd.GetNewSgwCTeid(ctx, gw.NetworkId)
+	if err != nil {
+		return 0, err
+	}
+
+	// TODO: remove logs after feature is tested
+	glog.Infof("Sgw C Teid generated by FegRelay: %s", teid)
```
Comment (emakeev) [inline feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/s8_relay.go]: maybe just glog.V(1).Infof ...?

File: feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/s8_relay.go [hunk 2]
```diff
@@ -98,3 +111,24 @@ func (s S8RelayRouter) getS8Client(c context.Context, imsi string) (protos.S8Pro
...
+
+	teidUint32, err := strconv.ParseUint(teid, 10, 32)
+	if err != nil {
+		return 0, err
+	}
+	return uint32(teidUint32), nil
```
Comment (emakeev) [inline feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/s8_relay.go]: I think, you can just get rid of this 'if': return uint32(teidUint32), err

**Review by emakeev — APPROVED** [review]
LGTM

File: feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/s8_relay.go (hunk 1 above)
Comment (uri200) [reply feg/cloud/go/services/feg_relay/gw_to_feg_relay/servicers/s8_relay.go]: sure

**Pull request merged**